src/parser/ts/index.ts

```diff
@@ -0,0 +1,528 @@
+  const res = iterator(route);
+  if (res !== undefined) {
+    resArr.push(res);
+  }
+  if (isSTSFunction(route)) {
+    let args: STSItem;
+    if(isMapFunction(route)) {
+      args = route.args[0];
+    } else {
+      args = route.args;
+    }
+    resArr.push(...walk(args, iterator));
+  }
+  return resArr;
+};
+
+export const getIndicesOfMultiaryFunctionArgs = (route: STSQueryItem[] | STSFunction): number[] => {
+  const responseArr = [];
+  const argsIndices = [];
+  let index = 0;
+  walk(route, obj => {
+    if (isMultiaryFunction(obj) && obj.args.length === 1) {
+      argsIndices.push(...obj.args);
+    } else if (isSTSReference(obj) && !hasIdsFilter(obj)) {
+      if (argsIndices.includes(obj)) {
+        responseArr.push(index);
+      }
+      index++;
+    }
+  });
+  return responseArr;
+};
+
+export const getServerFilters = (
+  route: STSQueryItem[] | STSFunction
+): StringMap[] => {
+  const responseArr: StringMap[] = [];
+  const filter = (_, __, parent) => isServerFilter(parent);
+  walk(route, obj => {
+    if (isSTSReference(obj) && !hasIdsFilter(obj)) {
+      const serverQuery = filterDeep(filterDeep(obj.query, filter), filter);
+      const flatFilters = flattenServerQueryFilters(serverQuery || []);
+      responseArr.push(flatFilters);
+    }
+  });
+  return responseArr;
+};
+
+export function flattenClientQueryFilters(filters: STSFilter[], path = []): STSClientFilter[] {
+  const res: STSClientFilter[] = [];
+  for(const filter of filters) {
+    if (isServerFilter(filter) && isSTSFilterArr(filter.value)) {
+      res.push(
+        ...flattenClientQueryFilters(filter.value, [...path, filter.path])
+      )
+    } else {
+      res.push({
+        path: [...path, filter.path].join('.'),
+        value: filter.value,
+        filter: filter.filter
+      } as STSClientFilter)
+    }
+  }
+  return res;
+}
+
+export const getClientFilters = (route: STSQueryItem[] | STSFunction): STSClientFilter[][] => {
+  const responseArr: STSClientFilter[][] = [];
+  const filter = (_, __, parent) => isClientFilter(parent) || isArray(parent.value);
+  walk(route, obj => {
+    if (isSTSReference(obj) && !hasIdsFilter(obj)) {
+      const clientQuery = filterDeep(filterDeep(obj.query, filter) || [], filter);
+      const flatFilters = flattenClientQueryFilters(clientQuery || []);
+      responseArr.push(flatFilters);
+    }
+  });
+  return responseArr;
+};
+
+function reverseSTSFilter({ path, filter, value }: STSFilter) {
+  return `${path}${filter}${stringifyValue(value)}`;
+};
+
+function stringifyValue(val: STSValue, wrap: boolean = true) {
+  if(isArray(val)) {
+    const items = (val as any).map(item => {
+      return isSTSFilter(item) ? reverseSTSFilter(item) : stringifyValue(item)
+    });
+    const joinedStr = items.join(', ');
+    if (wrap) {
+      return isSTSFilterArr(val) ? `{${joinedStr}}`: `[${joinedStr}]`;
+    }
+    return joinedStr;
+  }
+  return JSON.stringify(val);
+}
+
+export const reverse = (
```

Comment:
Nice one 👍 

But can we have a different name here? Smthig like `toSyntheticQuery` mb?

Reply:
https://stackoverflow.com/questions/148857/what-is-the-opposite-of-parse
composeSyntheticQuery maybe?

Reply:
`composeSyntheticQuery` sounds good